File: problems/Python/Arrays/Easy/MaximumSumOfSubArray.py
def solution(nums):
    maxSum = nums[0]
    currentSum = nums[0]

    for i in range(1, len(nums)):
        currentSum = currentSum if currentSum + nums[i] < currentSum else currentSum + nums[i]
        maxSum = maxSum if maxSum > currentSum else currentSum

    return maxSum

nums = [-2,1,-3,4,-1,2,1,-5,4]
print(solution(nums))

File: problems/Java/Arrays/Easy/MaximumSumOfSubArray.java
package problems.Java.Arrays;

public class MaximumSumOfSubArray {

    public  static int Solution(int[] nums){
        int currentSum = nums[0];
        int maxSum = nums[0];

        for(int i = 1; i < nums.length; i++){
            currentSum = Math.max(nums[i], currentSum + nums[i]);
            maxSum = Math.max(maxSum, currentSum);
        }

        return maxSum;
    }

    public static void main(String[] args){
        int[] arr = {1};
        System.out.print("Maximum Sum Of Arrays : "+ Solution(arr));
    }
}
